samples/extensions/raytracing_basic/raytracing_basic.cpp

```diff
@@ -238,266 +136,383 @@ uint64_t RaytracingBasic::get_buffer_device_address(VkBuffer buffer)
+	    get_device(),
+	    acceleration_structure_build_sizes_info.accelerationStructureSize,
+	    VK_BUFFER_USAGE_ACCELERATION_STRUCTURE_STORAGE_BIT_KHR,
+	    VMA_MEMORY_USAGE_GPU_ONLY);
+
+	// Create the acceleration structure
+	VkAccelerationStructureCreateInfoKHR acceleration_structure_create_info{};
+	acceleration_structure_create_info.sType  = VK_STRUCTURE_TYPE_ACCELERATION_STRUCTURE_CREATE_INFO_KHR;
+	acceleration_structure_create_info.buffer = bottom_level_acceleration_structure.buffer->get_handle();
+	acceleration_structure_create_info.size   = acceleration_structure_build_sizes_info.accelerationStructureSize;
+	acceleration_structure_create_info.type   = VK_ACCELERATION_STRUCTURE_TYPE_BOTTOM_LEVEL_KHR;
+	vkCreateAccelerationStructureKHR(device->get_handle(), &acceleration_structure_create_info, nullptr, &bottom_level_acceleration_structure.handle);
+
+	// The actual build process starts here
+
+	// Create a scratch buffer as a temporary storage for the acceleration structure build
+	ScratchBuffer scratch_buffer = create_scratch_buffer(acceleration_structure_build_sizes_info.buildScratchSize);
+
+	VkAccelerationStructureBuildGeometryInfoKHR acceleration_build_geometry_info{};
+	acceleration_build_geometry_info.sType                     = VK_STRUCTURE_TYPE_ACCELERATION_STRUCTURE_BUILD_GEOMETRY_INFO_KHR;
+	acceleration_build_geometry_info.type                      = VK_ACCELERATION_STRUCTURE_TYPE_BOTTOM_LEVEL_KHR;
+	acceleration_build_geometry_info.flags                     = VK_BUILD_ACCELERATION_STRUCTURE_PREFER_FAST_TRACE_BIT_KHR;
+	acceleration_build_geometry_info.mode                      = VK_BUILD_ACCELERATION_STRUCTURE_MODE_BUILD_KHR;
+	acceleration_build_geometry_info.dstAccelerationStructure  = bottom_level_acceleration_structure.handle;
+	acceleration_build_geometry_info.geometryCount             = 1;
+	acceleration_build_geometry_info.pGeometries               = &acceleration_structure_geometry;
+	acceleration_build_geometry_info.scratchData.deviceAddress = scratch_buffer.device_address;
+
+	VkAccelerationStructureBuildRangeInfoKHR acceleration_structure_build_range_info;
+	acceleration_structure_build_range_info.primitiveCount                                           = 1;
+	acceleration_structure_build_range_info.primitiveOffset                                          = 0;
+	acceleration_structure_build_range_info.firstVertex                                              = 0;
+	acceleration_structure_build_range_info.transformOffset                                          = 0;
+	std::vector<VkAccelerationStructureBuildRangeInfoKHR *> acceleration_build_structure_range_infos = {&acceleration_structure_build_range_info};
+
+	// Accelerations structures can be build either on the host or the device, with the former only possible if the implementation supports it
+	if (acceleration_structure_features.accelerationStructureHostCommands)
+	{
+		// Implementation supports building acceleration structure building on host
```

Comment:
Is there any good reason to prefer building on the host (which is optional), instead of just using device build all the time?

Reply:
Not really. Having that path in the if-part may be a bit misleading. Will switch that around.

Reply:
Might be nice to have it controlled from the outside (e.g. commandline) so people can compare build time and quality.

Reply:
I don't think there is a point for such a basic sample. I'd just keep it simple and only use the guaranteed device path in this sample.